### PR TITLE
p2p/protocols, swarm/network/stream, swarm/storage: clean debug changes

### PR DESCRIPTION
**Pull request by janos**
p2p/protocols, swarm/network/stream, swarm/storage: clean debug changes

Branch swarm-network-rewrite-syncer-test contains a number of
changes related to swarm/network/stream package debugging.
This change removes this changes and sets changed variables to
the ones in swarm-network-rewrite-syncer branch.

### CODE REVIEW

File: swarm/network/stream/delivery.go [hunk 1]
```diff
@@ -224,21 +215,17 @@ func (d *Delivery) processReceivedChunks() {
 			}
 			// go func() {
```
Comment (zelig) [inline swarm/network/stream/delivery.go]: remove comments
 I am not sure if this should be in a go routine at all, if storage cannot keep up with network there is bigger problem here. maybe there should be a just the buffer on receiveC and a select  on line 184 with default dropping the peer or without select 
a select  on line 184 with default dropping the peer or without select just blocking the network protocol until the buffer is freed.

File: swarm/network/stream/delivery.go [hunk 2]
```diff
@@ -224,21 +215,17 @@ func (d *Delivery) processReceivedChunks() {
...
 			chunk.SData = req.SData
-			log.Error("received delivery", "peer", req.peer.ID(), "hash", chunk.Key.Hex())
 			d.db.Put(chunk)
-			log.Error("put to db", "peer", req.peer.ID(), "hash", chunk.Key.Hex())
 			chunk.WaitToStore()
```
Comment (zelig) [inline swarm/network/stream/delivery.go]: better just put this wait and release in a go routine, but no timeout needed

File: swarm/network/stream/messages.go [hunk 3]
```diff
@@ -172,18 +169,18 @@ func (p *Peer) handleOfferedHashesMsg(req *OfferedHashesMsg) error {
 	go func() {
 		select {
 		case <-time.After(1 * time.Second):
```
Comment (zelig) [inline swarm/network/stream/messages.go]: this must be a longer timeout if at all needed

File: swarm/network/stream/peer.go [hunk 4]
```diff
@@ -101,10 +101,10 @@ func (p *Peer) SendOfferedHashes(s *server, f, t uint64) error {
 		Stream:        s.stream,
 		Key:           s.key,
 	}
-	log.Error("Swarm syncer offer batch", "peer", p.ID(), "stream", s.stream, "key", s.key, "len", len(hashes), "from", from, "to", to)
+	log.Trace("Swarm syncer offer batch", "peer", p.ID(), "stream", s.stream, "key", s.key, "len", len(hashes), "from", from, "to", to)
 	for i := 0; i < len(hashes); i += HashSize {
 		hash := hashes[i : i+HashSize]
-		log.Error("Swarm syncer offer hash", "peer", p.ID(), "stream", s.stream, "hash", storage.Key(hash).Hex(), "len", len(hashes), "from", from, "to", to)
+		log.Trace("Swarm syncer offer hash", "peer", p.ID(), "stream", s.stream, "hash", storage.Key(hash).Hex(), "len", len(hashes), "from", from, "to", to)
```
Comment (zelig) [inline swarm/network/stream/peer.go]: this should be removed

File: swarm/network/stream/syncer_test.go [hunk 5]
```diff
@@ -196,15 +188,15 @@ func testSyncBetweenNodes(t *testing.T, nodes, conns, chunkCount int, skipCheck
 				if err == storage.ErrFetching {
 					<-chunk.ReqC
 				} else if err != nil {
-					log.Error("not found", "index", i, "origin", j, "key", key.Hex(), "err", err)
+					log.Debug("not found", "index", i, "origin", j, "key", key.Hex(), "err", err)
 					continue
 				}
 				// needed for leveldb not to be closed?
```
Comment (zelig) [inline swarm/network/stream/syncer_test.go]: remove these two lines

File: swarm/storage/dbstore.go [hunk 6]
```diff
@@ -726,19 +686,14 @@ func (s *DbStore) get(key Key) (chunk *Chunk, err error) {
 			hash := hasher.Sum(nil)
 
 			if !bytes.Equal(hash, key) {
-				log.Error(fmt.Sprintf("Apparent key/hash mismatch. Hash %x, key %v", hash, key[:]))
+				log.Trace(fmt.Sprintf("Apparent key/hash mismatch. Hash %x, key %v", hash, key[:]))
```
Comment (zelig) [inline swarm/storage/dbstore.go]: these should be errors actually

File: swarm/storage/dbstore.go [hunk 7]
```diff
@@ -804,7 +759,7 @@ func (s *DbStore) SyncIterator(since uint64, until uint64, po uint8, f func(Key,
 	defer it.Release()
 	it.Seek(sincekey)
 
-	for it.Next() {
+	for it.Valid() {
```
Comment (zelig) [inline swarm/storage/dbstore.go]: why put this back? i thought we said this was obsolete

**Review by zelig — CHANGES_REQUESTED** [review]

File: swarm/storage/dbstore.go (hunk 7 above)
Comment (janos) [reply swarm/storage/dbstore.go]: Test TestIterator fails if this is used https://github.com/ethersphere/go-ethereum/pull/216/commits/31124f65bf7ef2bb921f2168b6699a199c0c0d1e. It does not break other tests or benchmarks.

File: swarm/storage/dbstore.go (hunk 6 above)
Comment (janos) [reply swarm/storage/dbstore.go]: The Trace log function is but back from https://github.com/ethersphere/go-ethereum/blob/swarm-network-rewrite-syncer/swarm/storage/dbstore.go#L694. I was not aware that it was corrected in swarm-network-rewrite-syncer-test, I will change it.

**Comment by janos** [comment]
Actually, all swarm/network/stream benchmarks pass successfully on this branch even without https://github.com/ethersphere/go-ethereum/pull/217/commits/b2b9a9aa81bc44234230ef6450300d5bed754145 changes.

**Comment by janos** [comment]
@zelig the code is updated by your comments

**Comment by zelig** [comment]
the synciterator is not what we want in the network right? we need left open right closed `(]` intervals

**Comment by janos** [comment]
Shouldn't it be `[)`?

**Comment by zelig** [comment]
ok lets keep it this way, but should test carefully esp when we have intervals

**Pull request merged**